test/system/api_keys_test.rb

```diff
@@ -153,6 +153,21 @@ class ApiKeysTest < ApplicationSystemTestCase
     assert_predicate @user.api_keys.last, :mfa_enabled?
   end
 
+  test "creating new api key with expiration" do
```

Comment:
Can you update expiration on a key? Maybe a test for that.

Reply:
IMO we don't want to allow updating expiration after the fact, and you should generate a new key if you need a new expiration.

Reply:
It's disabled in the UI and also in the controller, will add a functional test for the latter